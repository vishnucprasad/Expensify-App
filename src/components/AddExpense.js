import React from 'react';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class AddExpense extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div className="content">
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                    </div>
                </div>
                <div className="content-container card">
                    <ExpenseForm onSubmit={this.onSubmit} />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpense);