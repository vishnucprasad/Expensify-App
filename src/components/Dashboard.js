import React from 'react';
import ExpenseList from './ExpenseList';

const Dashboard = () => (
    <div>
        <ExpenseList />
    </div>
);

export default Dashboard;