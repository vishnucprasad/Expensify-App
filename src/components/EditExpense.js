import React from 'react';

const EditExpense = () => (
    <div>
        This is from my edit expense component
    </div>
);

export default EditExpense;