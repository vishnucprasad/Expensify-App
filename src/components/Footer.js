import React from 'react';

const Footer = () => (
    <div className="footer">
        <div className="content-container">
            <p className="footer__text">&copy; 2021 Expensify. All rights reserved | Developed by <a href="https://vishnucprasad.github.io" target="_blank">Vishnu C Prasad</a></p>
        </div>
    </div>
);

export default Footer;