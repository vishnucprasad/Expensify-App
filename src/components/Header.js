import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';
import MySwal, { Toast } from '../swal/swal';

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Expensify</h1>
                </Link>
                <button
                    onClick={() => {
                        MySwal.fire({
                            title: 'Are you sure you want to logout?',
                            icon: 'question',
                            showCancelButton: true,
                            confirmButtonText: 'Logout',
                            reverseButtons: true
                        }).then((result) => {
                            if (result.isConfirmed) {
                                startLogout().then(() => {
                                    Toast.fire({
                                        icon: 'success',
                                        title: 'Logged out successfully'
                                    });
                                });
                            }
                        });
                    }}
                    className="button button--link">Logout</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);