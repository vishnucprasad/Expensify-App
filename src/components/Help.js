import React from 'react';

const Help = () => (
    <div>
        This is from my help component
    </div>
);

export default Help;