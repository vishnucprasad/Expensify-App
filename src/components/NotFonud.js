import React from 'react';
import { Link } from 'react-router-dom';

const NotFonud = () => (
    <div>
        404 - <Link to="/" >Go to home</Link>
    </div>
);

export default NotFonud;