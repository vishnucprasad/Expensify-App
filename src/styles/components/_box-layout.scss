.box-layout {
  align-items: center;
  background: url("/images/bg.jpg");
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
}

.box-layout__box {
  background-color: $glass-white-1;
  backdrop-filter: blur(20px);
  border-radius: 8px;
  box-shadow: $card-shadow;
  padding: $xl-size $l-size;
  text-align: center;
  width: 40rem;
  color: $white;
}

.box-layout__title {
  line-height: 1;
  margin: 0 0 $m-size 0;
}
