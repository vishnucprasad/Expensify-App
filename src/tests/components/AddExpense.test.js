import React from 'react';
import { shallow } from 'enzyme';
import { AddExpense } from '../../components/AddExpense';
import expenses from '../fixtures/expenses';

let onSubmit, history, wrapper;

beforeEach(() => {
    onSubmit = jest.fn();
    history = { push: jest.fn() };
    wrapper = shallow(<AddExpense onSubmit={onSubmit} history={history} />);
});

test('should render AddExpense correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(onSubmit).toHaveBeenLastCalledWith(expenses[1]);
    expect(wrapper).toMatchSnapshot();
});