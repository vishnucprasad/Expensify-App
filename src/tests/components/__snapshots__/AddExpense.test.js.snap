// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle onSubmit 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;

exports[`should render AddExpense correctly 1`] = `
<div>
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
